### PR TITLE
fixed test case getByConcept

### PR DESCRIPTION
**Pull request by dilekamadushan**
TRUNK - 5212 Add test case for ProgramWorkFlowService.getProgramsByConcept
I added a test case to test the function getProgramByConcept

 Checklist: I completed these to help reviewers :)

My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

I have **added tests** to cover my changes. 

 write tests and add them to this commit `git add . && git commit --amend`
I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by dkayiwa** [comment]
Did you get a chance to look at this? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

File: api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java [hunk 1]
```diff
@@ -56,6 +57,8 @@
 	protected EncounterService encounterService = null;
 	
 	protected ConceptService cs = null;
+
+	protected ProgramWorkflowDAO dao = null;
```
Comment (wluyima) [inline api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java]: You should be testing ProgramWorkflowService and not DAO, so you will need to call the service and not the DAO

Comment (dilekamadushan) [reply api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java]: @dkayiwa I will go through it again. Thanks.

**Comment by dilekamadushan** [comment]
@wluyima I modified the code to replace ProgramWorkFlowDAO with ProgramWorkFlowService

File: api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java [hunk 2]
```diff
@@ -67,6 +68,9 @@ public void runBeforeEachTest() {
 			encounterService = Context.getEncounterService();
 			cs = Context.getConceptService();
 		}
+
+        // taking dao from the spring application context object
+        pws = (ProgramWorkflowService) applicationContext.getBean("programWorkflowDAO");
```
Comment (wluyima) [inline api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java]: You should be testing ProgramWorkflowService and not the DAO

Comment (dilekamadushan) [reply api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java]: @wluyima  I will correct this one. Thanks

**Comment by dilekamadushan** [comment]
@wluyima I corrected the code on your latest reviews. Thanks..Feel free to tell me what is to left to be done.Thanks

**Comment by dkayiwa** [comment]
@dilekamadushan can you squash these commits into one as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by dilekamadushan** [comment]
@dkayiwa I squashed the commits as much as possible.

**Pull request closed without merging**